pkgs/tools/filesystems/mergerfs/default.nix

```diff
@@ -1,18 +1,19 @@
-{ stdenv, fetchgit, fuse, pkgconfig, which, attr, pandoc, git }:
+{ stdenv, fetchgit, fuse, pkgconfig, which, attr, pandoc, git, autoconf, automake, libiconv, libtool, gettext }:
 
 stdenv.mkDerivation rec {
   name = "mergerfs-${version}";
-  version = "2.16.1";
+  version = "2.22.1";
 
   # not using fetchFromGitHub because of changelog being built with git log
   src = fetchgit {
     url = "https://github.com/trapexit/mergerfs";
     rev = "refs/tags/${version}";
-    sha256 = "12fqgk54fnnibqiq82p4g2k6qnw3iy6dd64csmlf73yi67za5iwf";
+    sha256 = "1rw9jxzryn687ivfis44aaflqhk77arm71q8hzlrs8lnjk83fn20";
     deepClone = true;
+    leaveDotGit = true;
   };
 
-  buildInputs = [ fuse pkgconfig which attr pandoc git ];
+  buildInputs = [ fuse pkgconfig which attr pandoc git autoconf automake libiconv libtool gettext ];
```

Comment:
I realized that since 2.22.0 `mergerfs` comes bundled with `fuse` which is why Autoconf et al. are now necessary. This however means that the `autoreconfHook` cannot be applied here because `mergerfs` itself does not use it as it is only needed for `fuse`.